src/test/groovy/org/prebid/server/functional/tests/SeatNonBidSpec.groovy

```diff
@@ -0,0 +1,251 @@
+
+        and: "PBS response shouldn't contain debug"
+        assert !response?.ext?.debug
+    }
+
+    def "PBS shouldn't populate seatNonBid when returnAllBidStatus=false and bidder didn't bid for any reason"() {
+        given: "Default bid request with disabled returnAllBidStatus"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.returnAllBidStatus = false
+        }
+
+        and: "Default bidder response without bids"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid = []
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse, PBSUtils.getRandomElement(HttpStatusCode.values() as List))
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response shouldn't contain seatNonBid"
+        assert !response.ext.seatnonbid
+        assert !response.seatbid
+    }
+
+    def "PBS shouldn't populate seatNonBid when debug=0 and requested bidder didn't bid for any reason"() {
+        given: "Default bid request with returnAllBidStatus and debug, test"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.returnAllBidStatus = false
+            ext.prebid.debug = 0
+        }
+
+        and: "Default bidder response without bids"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid = []
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse, PBSUtils.getRandomElement(HttpStatusCode.values() as List))
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response shouldn't contain seatNonBid for called bidder"
+        assert !response.ext.seatnonbid
+    }
+
+    def "PBS shouldn't populate seatNonBid with successful bids"() {
+        given: "Default bid request with enabled returnAllBidStatus"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.returnAllBidStatus = true
+        }
+
+        and: "Default bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response shouldn't contain seatNonBid and should contain seatbid"
+        assert response.ext.seatnonbid.nonBid.size() == 0
+        assert response.seatbid
+    }
+
+    def "PBS should populate seatNonBid when rejected due to timeout"() {
+        given: "PBS config with min time-out"
+        def pbsService = pbsServiceFactory.getService(
+                ["auction.biddertmax.max": "1",
```

Comment:
I don't think there is a need to change max timeout at all. You should only be worried about minimal.